### PR TITLE
Usergrant request cap req cap info

### PR DESCRIPTION
**Pull request by Karthick-Somasundaresan**
## What

Added support for usergrants.request
## Why

This Firebolt API needs to be added to support newer version
## How

We have added new API to support this
## Test

By calling usergrants.request it should get the grants from the user according to the usergrant policy.
## Checklist

- [x] I have self-reviewed this PR
- [ ] I have added tests that prove the feature works or the fix is effective

### CODE REVIEW

File: core/launcher/src/manager/app_launcher.rs [hunk 1]
```diff
@@ -608,9 +608,13 @@ impl AppLauncher {
             }
         }
 
-        let modified_url = Self::get_modified_url(&manifest, &sessionid);
-        debug!("modified url with sessionid : {:?}", modified_url);
-        Ok(modified_url)
+        if !sessionid.is_empty() {
```
Comment (satlead) [inline core/launcher/src/manager/app_launcher.rs:611]: needs merging with latest code.

File: core/main/src/service/apps/delegated_launcher_handler.rs [hunk 2]
```diff
@@ -603,7 +604,12 @@ impl DelegatedLauncherHandler {
                     false
                 }
             })
-            .filter(|perm| ps.cap_state.generic.check_all(&vec![perm.clone()]).is_ok())
```
Comment (satlead) [inline core/main/src/service/apps/delegated_launcher_handler.rs:606]: why only supported is checked here?

File: core/sdk/src/api/firebolt/fb_capabilities.rs [hunk 3]
```diff
@@ -84,6 +111,23 @@ impl FireboltCap {
         None
     }
 
+    pub fn parse_long(cap: String) -> Option<FireboltCap> {
+        let pattern = r"^xrn:firebolt:capability:([a-z0-9\\-]+)((:[a-z0-9\\-]+)?)$";
```
Comment (satlead) [inline core/sdk/src/api/firebolt/fb_capabilities.rs:115]: there is an existing parses method can we update that?

File: core/launcher/src/manager/app_launcher.rs (hunk 1 above)
Comment (Karthick-Somasundaresan) [reply core/launcher/src/manager/app_launcher.rs]: Yes, the branch was behind the main. Now Merged it.

File: core/main/src/service/apps/delegated_launcher_handler.rs (hunk 2 above)
Comment (Karthick-Somasundaresan) [reply core/main/src/service/apps/delegated_launcher_handler.rs:607]: This is to get the user's consent before launching the app. It is not going to use the cap immediately. So it is enough to check if the cap is supported. Only at the time of invocation, we have check if it is available.

File: core/sdk/src/api/firebolt/fb_capabilities.rs (hunk 3 above)
Comment (Karthick-Somasundaresan) [reply core/sdk/src/api/firebolt/fb_capabilities.rs:104]: Updated

**Review by satlead — APPROVED** [review]

**Review by maggie98choy — APPROVED** [review]

File: core/main/src/firebolt/handlers/capabilities_rpc.rs [hunk 4]
```diff
@@ -222,15 +236,22 @@ impl CapabilityServer for CapabilityImpl {
         ctx: CallContext,
         grants: CapRequestRpcRequest,
     ) -> RpcResult<Vec<CapabilityInfo>> {
-        let req_list: Vec<FireboltPermission> = grants.clone().into();
-        let permitted_result =
-            PermissionHandler::check_permitted(&self.state, &ctx.app_id, &req_list).await;
+        let fb_perms: Vec<FireboltPermission> = grants.clone().into();
```
Comment (satlead) [inline core/main/src/firebolt/handlers/capabilities_rpc.rs:239]: Ref

**Pull request merged**